src/Symfony/Component/Notifier/Bridge/Twitter/TwitterTransport.php

```diff
@@ -0,0 +1,297 @@
+        ksort($sign);
+
+        $oauth['oauth_signature'] = base64_encode(hash_hmac(
+            'sha1',
+            implode('&', array_map('rawurlencode', [
+                $method,
+                $url,
+                implode('&', array_map(fn ($k) => rawurlencode($k).'='.rawurlencode($sign[$k]), array_keys($sign))),
+            ])),
+            rawurlencode($this->apiSecret).'&'.rawurlencode($this->accessSecret),
+            true
+        ));
+
+        $options['headers'][] = 'Authorization: OAuth '.implode(', ', array_map(fn ($k) => $k.'="'.rawurlencode($oauth[$k]).'"', array_keys($oauth)));
+
+        return $this->client->request($method, $url, $options);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $options['text'] = $message->getSubject();
+
+        if (isset($options['attach'])) {
+            $options['media']['media_ids'] = $this->uploadMedia($options['attach']);
+            unset($options['attach']);
+        }
+
+        $response = $this->request('POST', '/2/tweets', ['json' => $options]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote Twitter server.', $response, 0, $e);
+        }
+
+        $result = $response->toArray(false);
+
+        if (400 <= $statusCode) {
+            throw new TransportException($result['title'].': '.($result['errors'][0]['message'] ?? $result['detail']), $response);
+        }
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId($result['data']['id']);
+
+        return $sentMessage;
+    }
+
+    /**
+     * @param array<array{file: File, alt: string, subtitles: File|null, category: string|null, owners: string[]}> $media
+     */
+    private function uploadMedia(array $media): array
+    {
+        $i = 0;
+        $pool = [];
+        $mediaIds = [];
+
+        foreach ($media as [
+            'file' => $file,
+            'alt' => $alt,
+            'subtitles' => $subtitles,
+            'category' => $category,
+            'owners' => $extraOwners,
+        ]) {
+            $query = [
+                'command' => 'INIT',
+                'total_bytes' => $file->getSize(),
+                'media_type' => $file->getContentType(),
+            ];
+            if ($category) {
+                $query['media_category'] = $category;
+            }
+            if ($extraOwners) {
+                $query['additional_owners'] = implode(',', $extraOwners);
+            }
+
+            $pool[++$i] = $this->request('POST', '/1.1/media/upload.json', [
```

Comment:
```suggestion
            $query = [
                'command' => 'INIT',
                'total_bytes' => $file->getSize(),
                'media_type' => $file->getContentType(),
            ];

            if ($category) {
                $query['media_category'] = $category;
            }

            if ($extraOwners) {
                $query['additional_owners'] = implode(',', $extraOwners);
            }

            $pool[++$i] = $this->request('POST', '/1.1/media/upload.json', [
```

Adding those new lines would make this to easier to read. Also notice that empty line before  `$pool[++$i] ...` so imho those `if` statements should have those too.